### DISCUSSION TITLE
Is it possible to use multimodal llava with llama.cpp?

### DISCUSSION THREAD

**Discussion by JasonS09** [oobabooga/text-generation-webui#4502] [Q&A]
Since I have a 4gb vram card, I use .gguf models. [Here I found a ggml version of llava1.5-7b](https://huggingface.co/mys/ggml_llava-v1.5-7b). I run the webui with llama.cpp and --cpu --numa flags. However, when I was trying to run it, even though chatting works fine, when I share an image, the program hangs at "_Is typing..._". It will never produce an output even after more than 10 minutes. I'm forced to stop the generation to continue the chat, defeating the purpose of multimodal use. I have a system with 8gb ram. Is it too little to run a multimodal model?

**Comment by BetaDoggo** [comment]
It's not possible right now because the llama-cpp-python library that the webui uses doesn't support it yet. It's a pinned issue in their repo so there's interest: https://github.com/abetlen/llama-cpp-python/issues/813

Once it does get support it will take some additional work to make it work with the existing extension as well.
